### ISSUE TITLE
[BUG] b2b_logic: wrong transport for transfer with "hold"

### ISSUE THREAD

**Issue by ovidiusas** [OpenSIPS/opensips#3138]
While performing a transfer using "hold" parameter, the transfer fails because the INVITE to the transferee is sent over UDP instead of the specified transport in URI.
Before initiating the transfer, a new b2b client is created:
`b2b_client_new($var(client_name), $var(refer_to),,, "sip:user@domain.com:5061;transport=tls", $avp(hdrs), $avp(hdrs_body));`
Then a bridge is requested:
`$var(bridge_flags) = "notify,rollback-failed,hold,no-late-sdp";`
`b2b_bridge("peer", $var(client_name),, $var(bridge_flags));`
The call to the transferee is failing because it is sent over UDP instead of TLS.

The transfer is successful if "hold" is removed from the bridge flags.
Here's the call flow diagram:
```
caller         b2b       transferor    transferee
  |----INVITE-->|             |             |
  |             |---INVITE--->|             |
  |             |<--200-------|             |
  |<---200------|             |             |
  |----ACK----->|             |             |
  |             |---ACK------>|             |
  |             |<--REFER-----|             |
  |             |---202------>|             |
  |<-reINVITE---|                           |
  |----200----->|                           |
  |<---ACK------|                           |
  |             |-----------------INVITE--->|
                (sent over UDP insted of TLS)
```
OpenSIPS version: 3.4

**Comment by bogdan-iancu** [comment]
Hi @ovidiusas , checking the docs for [b2b_client_new](ttps://opensips.org/html/docs/modules/3.4.x/b2b_logic.html#func_b2b_client_new), I see that the parameter (where you have the URI with the TLS transport) is documented as :+1: 
```
adv_contact (string, optional) - Contact header to advertise in generated messages. 
```
So, being a contact in the request, not sure how it should influence the destination transport of the request.

**Comment by ovidiusas** [comment]
Forgot to mention that the `$var(refer_to)` is populated as `"sip:user@destination_domain.com:5061;user=phone;transport=tls"`.

**Comment by ovidiusas** [comment]
The issue here seems to be related to the socket being retrieved from the 200ok received from the caller.
As a hack, in logic.c in b2bl_new_client I set the `ci->send_sock` to `NULL` and the INVITE to the transferee was sent correctly over tls.

**Comment by ovidiusas** [comment]
Here it is:
```
diff --git a/modules/b2b_logic/logic.c b/modules/b2b_logic/logic.c
index 546473e5d..c910031b9 100644
--- a/modules/b2b_logic/logic.c
+++ b/modules/b2b_logic/logic.c
@@ -515,8 +515,17 @@ b2bl_entity_id_t *b2bl_new_client(client_info_t *ci, b2bl_tuple_t *tuple,
        b2bl_entity_id_t* entity;

        ci->method = method_invite;
-       ci->send_sock = msg ?
-               (msg->force_send_socket?msg->force_send_socket:msg->rcv.bind_address):NULL;
+       if (msg) {
+               LM_NOTICE("msg->first_line.type=[%d] msg->REQ_METHOD=[%d]\n", msg->first_line.type, msg->REQ_METHOD);
+               if (msg->force_send_socket) {
+                       LM_NOTICE("msg->force_send_socket=[%.*s]\n", msg->force_send_socket->name.len, msg->force_send_socket->name.s);
+               } else {
+                       LM_NOTICE("msg->rcv.bind_address=[%.*s]\n", msg->rcv.bind_address->name.len, msg->rcv.bind_address->name.s);
+               }
+       }
+       //ci->send_sock = msg ?
+       //      (msg->force_send_socket?msg->force_send_socket:msg->rcv.bind_address):NULL;
+       ci->send_sock = NULL;

        if (adv_ct) {
                ci->local_contact = *adv_ct;
```

**Comment by bogdan-iancu** [comment]
This is an interesting logical conflict. The original code was "seeding" any new b2b_client with the socket of the already existing entities. The idea behind is to keep the whole scenario (1) under the same socket (like when you have multiple UDP interfaces) and prevent jumping across different interfaces with the same proto and (2) also to keep it under the same protocol (if the new entities are not explicit about that).
And the this `send_sock` is used to inherit the socket. 
On the other hand, I agree that the new entities, if there are no other constraints, should obey any explicit proto in RURI.

Unfortunately this single existing mechanism (of `send_sock`) is not so flexible to make the difference between "suggestions" versus "enforcement" . Like the existing interface may "suggest" a socket/proto, but it should give prio to any explicit transport "enforcement" in URI. Or an forced interface (script level) is an "enforcement" of the socket/proto, that may override any "suggestion" of the transport to use in the RURI (like if no clear transport indication, we should assume UDP, still this may be override by the forced socket).

Bottom line, the `c->send_socke` is too weak/simple to pass all the flavors between sockets and uri transport.
